index.js

```diff
@@ -22,24 +24,24 @@ var Config = require('./lib/Config');
  * @return {Object}
  */
 module.exports = function(options) {
-  var config = new Config(options);
+  const config = new Config(options);
 
   if (!fs.existsSync(config.filename)) {
     debug('file ' + config.filename + ' does not exist');
     return config.isListForm ? [] : {};
   }
 
-  var results = traverse(config);
+  const results = traverse(config);
   debug('traversal complete', results);
 
   dedupeNonExistent(config.nonExistent);
   debug('deduped list of nonExistent partials: ', config.nonExistent);
 
-  var tree;
+  let tree;
   if (config.isListForm) {
     debug('list form of results requested');
 
-    tree = removeDups(results);
+    tree = Array.from(new Set(results));
```

Comment:
How does the runtime and space complexity compare to the old implementation?

Reply:
I used the run time of the tests suite as a quick sanity check while I worked on this and it didn't change significantly.

According to https://medium.com/@jakubsynowiec/unique-array-values-in-javascript-7c932682766c it should be quite a bit faster than the old code.

Reply:
That’s fair. I’ll back that part out once I get to my notebook.